src/osd/osd_types.h

```diff
@@ -1345,6 +1350,9 @@ struct pg_pool_t {
       stripe_width(0),
       expected_num_objects(0),
       fast_read(false),
+      compress_algorithm(Compressor::COMP_ALG_NONE),
+      compress_ratio(0),
+      compress_hint(0),
```

Comment:
Instead of making these pg_pool_t members, they can go in the key/value area with random option fields.  See pool_opts_t